pkg/apiserver/webhook/cloudprovideraccount_webhook.go

```diff
@@ -106,7 +106,10 @@ type CPAValidator struct {
 // Handle handles validator admission requests for CPA.
 func (v *CPAValidator) Handle(ctx context.Context, req admission.Request) admission.Response {
 	v.Log.V(1).Info("Received CPA admission webhook request", "Name", req.Name, "Operation", req.Operation)
-	if !cloud.GetControllerSyncStatusInstance().IsControllerSynced(cloud.ControllerTypeCPA) {
+	// Any operation on CPA CR is not allowed until CES is synced.
+	// Since CPA CR is parent of CES CR, deletion of CPA CR would invoke delete of CES CR,
+	// which is not allowed.
+	if !cloud.GetControllerSyncStatusInstance().IsControllerSynced(cloud.ControllerTypeCES) {
```

Comment:
This is added for all cases, add, delete.. Add should depend on CPA only and Delete should depend on CPA and CES

Reply:
Done. Updated the PR. Also updated the unit-test to check for the appropriate error message.